File: android/src/main/java/expo/modules/libsignal/SealedSenderRefs.kt
package expo.modules.libsignal

import expo.modules.kotlin.sharedobjects.SharedObject
import org.signal.libsignal.metadata.certificate.SenderCertificate
import org.signal.libsignal.metadata.certificate.ServerCertificate

class ServerCertificateRef(val cert: ServerCertificate) : SharedObject()

class SenderCertificateRef(val cert: SenderCertificate) : SharedObject()

File: ios/SealedSenderRefs.swift
import Foundation
import ExpoModulesCore
import LibSignalClient

final class ServerCertificateRef: SharedObject {
  let cert: ServerCertificate

  init(cert: ServerCertificate) {
    self.cert = cert
    super.init()
  }
}

final class SenderCertificateRef: SharedObject {
  let cert: SenderCertificate

  init(cert: SenderCertificate) {
    self.cert = cert
    super.init()
  }
}
